kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue

```diff
@@ -161,17 +186,18 @@
       },
       editUserHandler() {
         const payload = {
+          id: this.userid,
           username: this.username_new,
           full_name: this.fullName_new,
-          facility: this.facility,
+          kind: this.kind_new,
         };
-        this.updateUser(this.userid, payload, this.role_new);
+        this.updateUser(payload);
         // if logged in admin updates role to learner, redirect to learn page
         // Do SUPERUSER check, as it is theoretically possible for a DeviceAdmin
         // to have the same id as a regular user, as they are different models.
         if ((this.session_user_kind !== UserKinds.SUPERUSER) &&
           (Number(this.userid) === this.session_user_id)) {
-          if (this.role_new === UserKinds.LEARNER.toLowerCase()) {
+          if (this.kind_new === UserKinds.LEARNER.toLowerCase()) {
```

Comment:
have you check what the value is given to kind_new in `<select v-model="kind_new" id="user-role">` coz all option values are uppercase.